app/src/main/kotlin/com/simplemobiletools/calendar/pro/activities/MainActivity.kt

```diff
@@ -526,7 +526,11 @@ class MainActivity : SimpleActivity(), RefreshRecyclerViewListener {
     }
 
     private fun showFilterDialog() {
-        FilterEventTypesDialog(this) {
+        SelectEventTypesDialog(this, config.displayEventTypes) {
+            if (config.displayEventTypes != it) {
```

Comment:
I think we can skip the check here, or maybe add a "return" if nothing changed, depending on the desired behaviour